fairseq/sequence_generator.py

```diff
@@ -481,7 +506,7 @@ def _generate(
             new_cands_to_ignore, active_hypos = torch.topk(
                 active_mask, k=beam_size, dim=1, largest=False
             )
-
+            negative_active_hypos = active_hypos
```

Comment:
I don't think we need to maintain `negative_active_hypos` inside of `SequenceGenerator`, it can be done in `LexicallyConstrainedBeamSearch` instead.

In particular, it's set here to `active_hypos`, but we already call `self.search.update_constraints(active_hypos)` below (line 544), so the `LexicallyConstrainedBeamSearch` instance can just save/maintain this internally.

Reply:
Done.